youtube_dl/YoutubeDL.py

```diff
@@ -1524,7 +1524,7 @@ def dl(name, info):
                         fd.add_progress_hook(ph)
                     if self.params.get('verbose'):
                         self.to_stdout('[debug] Invoking downloader on %r' % info.get('url'))
-                    return fd.download(name, info)
+                    return fd.download(name, info, self.params.get("md5"))
```

Comment:
It `md5` flag should not go here as an arguments. Instead it should be extracted in particular downloader from `self.params`.